src/Language/PureScript/Docs/Convert.hs

```diff
@@ -38,6 +102,95 @@ convertModule m@(P.Module _ coms moduleName  _ _) =
     >>> augmentDeclarations
     >>> map addDefaultFixity
 
+updateReExports ::
+  (Functor m, Applicative m,
+  MonadState (Map P.ModuleName (P.Module, Module)) m) =>
+  P.Env ->
+  [P.ModuleName] ->
+  m ()
+updateReExports env order = do
+  void $ traverse go order
+  where
+  go mn = do
+    (ast, mdl) <- lookup' mn
+    mdl' <- expandReExports env ast mdl
+    modify (Map.insert mn (ast, mdl'))
+
+  lookup' key = do
+    v <- gets (Map.lookup key)
+    case v of
+      Just v' ->
+        pure v'
+      Nothing ->
+        P.internalError ("Language.PureScript.Docs.Convert: missing key: " ++
+                        P.runModuleName key)
+
+-- We require that modules have already been sorted (P.sortModules) in order to
+-- ensure that by the time we convert a particular module, all its dependencies
+-- have already been converted.
+expandReExports ::
+  (Functor m, Applicative m,
+  MonadState (Map P.ModuleName (P.Module, Module)) m) =>
+  P.Env ->
+  P.Module ->
+  Module ->
+  m Module
+expandReExports env ast@(P.Module _ _ mn' _ mexports) mdl = do
+  let exports = fromMaybe (err "exports were not elaborated") mexports
+  let reExportNames = mapMaybe takeReExport exports
```

Comment:
The information you're after isn't really accessible, as the renaming never takes the intermediate step, it always has multiply-re-exported values pointing at the original export locations.

What's actually the problem with that? You'll have to manually follow the chain of the links through the docs back to the original module if you want to get the definition's comments anyway, so this just takes those steps out of it...

Instead of grouping the re-exported values under a heading for each module, why don't we just lump them all under "Re-exported definitions" in general, and then show the original module info per re-exported member instead? That'd cut down on the number of headers at least, if that's one of the concerns.